test/unit/store/checkin-store.js

```diff
@@ -0,0 +1,81 @@
+import CheckinStore from '../../../.dist/store/checkin-store';
+import ModelStore from '../../../.dist/store/model-store';
+import ItemStore from '../../../.dist/store/item-store';
+import StudentStore from '../../../.dist/store/student-store';
+import { assert } from 'chai';
+import { addAction } from '../../util/database';
+
+describe('CheckinStore', () => {
+
+    let model;
+    let items = [];
+    let student;
+
+    before(() => {
+        return addAction('CLEAR_ALL_DATA').then(() => {
+            return addAction('NEW_MODEL', {
+                name: 'Resistor'
+            });
+        }).then(actionId => {
+            model = ModelStore.getModelByActionId(actionId);
+            return addAction('NEW_ITEM', {
+                modelAddress: model.address
+            });
+        }).then(actionId => {
+            items.push(ItemStore.getItemByActionId(actionId));
+            return addAction('NEW_ITEM', {
+                modelAddress: model.address
+            });
+        }).then(actionId => {
+            items.push(ItemStore.getItemByActionId(actionId));
+            return addAction('NEW_ITEM', {
+                modelAddress: model.address
+            });
+        }).then(actionId => {
+            items.push(ItemStore.getItemByActionId(actionId));
+            return addAction('NEW_ITEM', {
+                modelAddress: model.address
+            });
+        }).then(() => {
+            return addAction('NEW_STUDENT', {
+                id: '123456',
+                name: 'John von Neumann'
+            });
+        }).then(actionId => {
+            student = StudentStore.getStudentByActionId(actionId);
+            return addAction('NEW_CHECKOUT', {
+                studentId: student.id,
+                itemAddresses: [items[0].address, items[2].address]
+            });
+        });
+    });
+
+    it('should instantiate without any checkins', () => {
+        assert.lengthOf(CheckinStore.getCheckins(), 0);
+    });
+
+    it('should instantiate with any checkin errors', () => {
```

Comment:
Please change the description to "should instantiate with**out** any checkin errors", so its a touch more descriptive.